tests/ipsec/ipsec3hosts.pm

```diff
@@ -0,0 +1,126 @@
+        script_run("ping -c 1 $setup->{right_ip}");
+        $self->config_ipsec($ipsec_setting_left);
+        barrier_wait 'IPSEC_TUNNEL_MODE_SETUP_DONE';
+        assert_script_run("ping -c 8 $setup->{right_ip}");
+        barrier_wait 'IPSEC_SET_MTU_DONE';
+        assert_script_run("ping6 -s 1300 -c 8 $setup->{right_ip}");
+        barrier_wait 'IPSEC_TUNNEL_MODE_CHECK_DONE';
+        $self->{ipsec_mode} = "transport";
+        $self->config_ipsec($ipsec_setting_left);
+        barrier_wait 'IPSEC_TRANSPORT_MODE_SETUP_DONE';
+        assert_script_run("ping6 -s 1300 -c 8 $setup->{right_ip}");
+        barrier_wait 'IPSEC_TRANSPORT_MODE_CHECK_DONE';
+    }
+
+    if (get_var('IPSEC_SETUP') eq 'middle') {
+        my ($dev0, $dev1) = split("\n", iface(2));
+        assert_script_run("sysctl net.ipv6.conf.all.forwarding=1");
+        #assert_script_run("ip link set $dev0 up");
+        assert_script_run("ip link set $dev1 down");
+        $self->add_ipv6_addr(ip => $setup->{middle_ip_01}, dev => $dev0, plen => $self->get_net_prefix_len($setup->{middle_net_01}));
+        assert_script_run("ip link set $dev1 up");
+        $self->add_ipv6_addr(ip => $setup->{middle_ip_02}, dev => $dev1, plen => $self->get_net_prefix_len($setup->{middle_net_02}));
+        $self->check_ipv6_addr();
+        script_run("ping -c 1 $setup->{left_ip}");
+        script_run("ping -c 1 $setup->{right_ip}");
+        script_run("ip neighbor show");
+        barrier_wait 'IPSEC_IP_SETUP_DONE';
+        barrier_wait 'IPSEC_ROUTE_SETUP_DONE';
+        script_run("ip a");
+        script_run("ip -6 route");
+        barrier_wait 'IPSEC_TUNNEL_MODE_SETUP_DONE';
+        assert_script_run("ip l s mtu 1300 dev $dev1");
+        barrier_wait 'IPSEC_SET_MTU_DONE';
+        barrier_wait 'IPSEC_TUNNEL_MODE_CHECK_DONE';
+        barrier_wait 'IPSEC_TRANSPORT_MODE_SETUP_DONE';
+        barrier_wait 'IPSEC_TRANSPORT_MODE_CHECK_DONE';
+    }
+
+    if (get_var('IPSEC_SETUP') eq 'right') {
+        $self->add_ipv6_addr(ip => $setup->{right_ip}, plen => $self->get_net_prefix_len($setup->{right_net}));
+        $self->check_ipv6_addr();
+        barrier_wait 'IPSEC_IP_SETUP_DONE';
+        script_run("ping -c 1 $setup->{middle_ip_02}");
+        script_run("ip neighbor show");
+        $self->add_ipv6_route(dst => $setup->{left_ip}, via => $setup->{middle_ip_02});
+        barrier_wait 'IPSEC_ROUTE_SETUP_DONE';
+        script_run("ip a");
+        script_run("ip -6 route");
+        script_run("ping -c 1 $setup->{left_ip}");
+        $self->config_ipsec($ipsec_setting_right);
+        barrier_wait 'IPSEC_TUNNEL_MODE_SETUP_DONE';
+        assert_script_run("tcpdump -i eth0 esp -c 4");
+        barrier_wait 'IPSEC_SET_MTU_DONE';
+        barrier_wait 'IPSEC_TUNNEL_MODE_CHECK_DONE';
+        $self->{ipsec_mode} = "transport";
+        $self->config_ipsec($ipsec_setting_right);
+        barrier_wait 'IPSEC_TRANSPORT_MODE_SETUP_DONE';
+        assert_script_run("tcpdump -i eth0 esp -c 4");
```

Comment:
What is the point of using tcpdump in the test if we are not checking its output. Is just for debugging  purpose?

Reply:
Not debugging purpose, although not checking its output but this command still can make sure we get correct number of esp packet (icmp packet).

Reply:
nit: we could install tcpdump only on the node which is using it (e.g. ipsec_3hosts_right_node). But I'm ok with how it is and going to merge it.